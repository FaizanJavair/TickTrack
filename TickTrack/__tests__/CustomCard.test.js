import React from "react";
import renderer from "react-test-renderer";

import CustomCard from "../components/customCard";

it("Custom Card renders correctly", () => {
  const tree = renderer
    .create(
      <CustomCard
        temperature={"67"}
        description={"Something"}
        tempUnit={"Â°"}
        desc1={"79"}
        unit1={"%"}
        desc2={"42"}
        unit2={"%"}
        desc3={"2"}
        unit3={"m/s"}
        desc4={"8"}
        unit4={"hPa"}
        mainImg={`http://openweathermap.org/img/wn/01n@4x.png`}
        dt={"1221212"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
