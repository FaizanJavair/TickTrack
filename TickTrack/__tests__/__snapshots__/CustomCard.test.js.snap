// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Card renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 12,
      "flexDirection": "row",
      "marginBottom": "3%",
      "padding": "4%",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 3,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 45,
          "fontWeight": "100",
          "marginLeft": "2%",
          "marginTop": "4%",
        }
      }
    >
      67
      Â°
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 22,
          "fontWeight": "300",
          "marginBottom": "3%",
          "marginLeft": "2%",
          "marginTop": "3%",
          "textTransform": "capitalize",
        }
      }
    >
      Something
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginTop": "2%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": "2%",
            "marginLeft": "2%",
          }
        }
      >
        <Text />
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "300",
              "marginBottom": "2%",
              "marginEnd": "2%",
              "marginStart": "2%",
            }
          }
        >
          79
          %
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": "2%",
            "marginLeft": "2%",
          }
        }
      >
        <Text />
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "300",
              "marginBottom": "2%",
              "marginEnd": "2%",
              "marginStart": "2%",
            }
          }
        >
          42
          %
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginTop": "2%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": "2%",
            "marginLeft": "2%",
          }
        }
      >
        <Text />
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "300",
              "marginBottom": "2%",
              "marginEnd": "2%",
              "marginStart": "2%",
            }
          }
        >
          2
           km/h
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": "2%",
            "marginLeft": "2%",
          }
        }
      >
        <Text />
        <Text
          style={
            {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "300",
              "marginBottom": "2%",
              "marginEnd": "2%",
              "marginStart": "2%",
            }
          }
        >
          8
           hPa
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "width": "30%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "http://openweathermap.org/img/wn/01n@4x.png",
        }
      }
      style={
        {
          "height": 100,
          "tintColor": "#28282B",
          "width": 100,
        }
      }
    />
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "400",
          "marginBottom": "10%",
          "marginLeft": "22%",
        }
      }
    >
      1221212
    </Text>
  </View>
</View>
`;
